packages/core/src/repository/query/network-query.ts

```diff
@@ -0,0 +1,20 @@
+import { ParameterType } from '../../data';
+
+export abstract class NetworkQuery {
```

Comment:
I find weird the `NetworkQuery` class setup:
- is `abstract`, but it doesn't expose any abstract method. Remove `abstract`?
- is exported; but it's useless on it's own, as we wouldn't know to which HTTP method to map in the network data-source. So, either:
    - We don't export this `NetworkQuery`.
    - Or, we export it, but we add a public `method` property. In which case the base class will have an "open" `GET | POST | PUT | DELETE` union, while the subclasses already specify their specific method, e.g. `GetNetworkQuery.method = 'GET'`.

Maybe, (better?, we rename `NetworkQuery` to `BaseNetworkQuery`; and apart, we create a union type `export type NetworkQuery = GetNetworkQuery | PostNetworkQuery | PutNetworkQuery | DeleteNetworkQuery;` which can be used in `NetworkDataSource.getRequestWithParameters` which I understand was the reason of exporting the base class.